firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreMultiDbComponent.java

```diff
@@ -18,12 +18,19 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseAppLifecycleListener;
+import com.google.firebase.FirebaseOptions;
 import com.google.firebase.auth.internal.InternalAuthProvider;
 import java.util.HashMap;
 import java.util.Map;
 
 /** Multi-resource container for Firestore. */
-class FirestoreMultiDbComponent {
+class FirestoreMultiDbComponent implements FirebaseAppLifecycleListener {
+
+  public interface DeregisterCommand {
```

Comment:
`java.lang.Runnable` instead?

Reply:
That's a good idea.

Alternatively, you could just make an interface that contains `remove(String databaseId)` such that `FirestoreMultiDbComponent` could just implement it. This removes the extra capture.

In any case, the interface describing this behavior should not be nested in `FirestoreMultiDbComponent` because that means `FirebaseFirestore` still ends up having an awareness of that type. Instead nest the interface in `FirebaseFirestore`.

Something like this is what I mean:

```java
class FirebaseFiresore {
  // ...
  interface InstanceRegistry {
    void remove(@NonNull String databaseId);
  }
}
```

FirestoreMultiDbComponent can implement this interface directly.

Reply:
Done.